### PR TITLE
External CI Tag Build Updates for Three Repos

### PR DESCRIPTION
**Pull request by amd-jmacaran**
- Updating build flags for llvm-project to support another pipeline to work with aomp repos.
- Added support for rocMLIR component.
- Removed MIVisionX python dependency script and leveraged existing dependencies template.

### CODE REVIEW

**Review by yhuiYH — DISMISSED** [review]

File: .azuredevops/components/rocMLIR.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  default:
+    - cmake
+    - ninja-build
+
+jobs:
+- job: rocMLIR
+  variables:
+  - group: common
+  - template: /.azuredevops/variables-global.yml
+  pool: ${{ variables.HIGH_END_BUILD_POOL }}
```
Comment (alexxu-amd) [inline .azuredevops/components/rocMLIR.yml:19]: Do we need to change this to LOW_END_BUILD_POOL or cloud?

Comment (amd-jmacaran) [reply .azuredevops/components/rocMLIR.yml:19]: Good catch

**Review by alexxu-amd — APPROVED** [review]
LGTM!

**Review by abhimeda — APPROVED** [review]
Looks good!

**Review by yhuiYH — APPROVED** [review]

**Pull request merged**